### PR TITLE
fix Issue 20592 - [ICE] dmd/iasmgcc.d(332): Assertion failure

### PR DESCRIPTION
**Pull request by ibuclaw**
The fall through was unchecked.

### CODE REVIEW

File: src/dmd/iasmgcc.d [hunk 1]
```diff
@@ -402,7 +409,7 @@ unittest
         p.check(TOK.rightCurly);
 
         auto res = semanticAsm(toklist);
-        assert(res == 0 || expectError);
+        assert((res == 0 && !expectError) || (expectError && res != 0));
```
Comment (MoonlightSentinel) [inline src/dmd/iasmgcc.d]: You could shorten this to
```suggestion
        assert((res == 0) != expectError);
```

**Review by thewilsonator — APPROVED** [review]

**Review by WalterBright — APPROVED** [review]

**Comment by ibuclaw** [comment]
Just need to rebase as other tests have been added.
